src/fides/api/ctl/sql_models.py

```diff
@@ -353,6 +381,49 @@ def get_data_uses(self, include_parents: bool = True) -> set[str]:
         return self.collapse_data_uses(self.privacy_declarations or [], include_parents)
 
 
+class PrivacyDeclaration(Base):
+    """
+    The SQL model for a Privacy Declaration associated with a given System.
+    """
+
+    name = Column(String)  # labeled as Processing Activity in the UI
+    ### keep egress/ingress as JSON blobs as they have always been
+    egress = Column(JSON)
+    ingress = Column(JSON)
```

Comment:
honestly, i'm not sure. i just tried to preserve what was there to keep things simple for now. but may be a good thing to drill into before we merge -- perhaps @mfbrown  or @ThomasLaPiana know? or would someone else?

Reply:
Want to highlight my other comment that may have been missed, it seems like the data type is wrong here and both the ingress and egress should be`Column(ARRAY(String))` instead.

fides lang:
```
class PrivacyDeclaration(BaseModel):
    """
    The PrivacyDeclaration resource model.

    States a function of a system, and describes how it relates
    to the privacy data types.
    """
     ...
    egress: Optional[List[FidesKey]] = Field(
        description="The resources to which data is sent. Any `fides_key`s included in this list reference `DataFlow` entries in the `egress` array of any `System` resources to which this `PrivacyDeclaration` is applied."
    )
    ingress: Optional[List[FidesKey]] = Field(
        description="The resources from which data is received. Any `fides_key`s included in this list reference `DataFlow` entries in the `ingress` array of any `System` resources to which this `PrivacyDeclaration` is applied."
    )
```

Reply:
Vaguely remembering back (this was in August or September I think?) we probably put this as a placeholder and put the real version in Fideslang or vice versa.

Currently, Fideslang is correct and that `egress` and `ingress` are a Fides key for a system or dataset

There was also a `DataFlow` object we were going to add but I'm not sure if we ever did...this was work that was pretty specific to the system scanner and generating data maps

Reply:
thanks for the reminder @pattisdr, i had indeed missed that. and thanks for the additional context @ThomasLaPiana.

so from what i can tell, i'll plan to put this as a "soft reference" to a fides key, i.e. as a `Column(ARRAY(String))` field, at least just to be able to store the data in case it comes in via API in some case (not sure it's exposed via the UI at any point currently). we can refactor or remove if/when this becomes formalized further.

let me know if either of you see any issues with that!

Reply:
that works Adam